src/renderer/context/MultisigChainContext/MultisigChainContext.tsx

```diff
@@ -143,10 +156,14 @@ export const MultisigChainProvider = ({ children }: PropsWithChildren) => {
     });
 
     return () => {
-      unsubscribeMultisigs.forEach((unsubscribeEvent) => unsubscribeEvent());
-      Promise.all(unsubscribeEvents).then(() => console.info('unsubscribed from events'));
+      unsubscribeMultisigs.forEach((unsubscribe) => unsubscribe());
+      unsubscribeEvents.forEach((unsubscribe) => unsubscribe());
     };
-  }, [debouncedConnections, account]);
+  }, [
+    Object.values(debouncedConnections).filter((c) => c.connection.connectionStatus === ConnectionStatus.CONNECTED)
```

Comment:
I assume it's to much for dependency array 😊
Let's make it a variable before `useEffect`